dask_geopandas/io/arrow.py

```diff
@@ -0,0 +1,374 @@
+import copy
+import json
+import math
+from typing import TYPE_CHECKING
+
+from dask.base import compute_as_if_collection, tokenize
+from dask.dataframe.core import new_dd_object, Scalar
+from dask.highlevelgraph import HighLevelGraph
+from dask.layers import DataFrameIOLayer
+from dask.utils import natural_sort_key, apply
+
+from dask.dataframe.io.utils import _get_pyarrow_dtypes, _meta_from_dtypes
+
+import pandas as pd
+import geopandas
+import shapely.geometry
+
+from fsspec.core import get_fs_token_paths
+
+
+if TYPE_CHECKING:
+    import pyarrow
+
+
+def _update_meta_to_geodataframe(meta, schema_metadata):
+    """
+    Convert meta to a GeoDataFrame and update with potential GEO metadata
+    """
+    if schema_metadata and b"geo" in schema_metadata:
+        geo_meta = json.loads(schema_metadata[b"geo"])
+        geometry_column_name = geo_meta["primary_column"]
+        crs = geo_meta["columns"][geometry_column_name]["crs"]
+        geometry_columns = geo_meta["columns"]
+    else:
```

Comment:
If this is a TODO, maybe this should raise a `NotImplementedError` until that is in place?

Reply:
I wouldn't raise an error. Incorrectly read file can be fixed (like if you read geo-arrow parquet with vanilla dask.dataframe) but if we raise you'd need to explicilty read it with dask.dataframe. But I would maybe warn at least (though geopandas probably does that already?)

Reply:
Yes, geopandas will currently actually raise an error in that case. But, that error will only happen when computing, while it's probably good to already raise an error upfront.

With the current PR, we actually already run into an error in this case, because loading of the partitions bounds fails. But that's not a very useful error. 

Short term: will add a descriptive error and test for this.